catalog/core/catalog-core-directorymonitor/src/main/java/org/codice/ddf/catalog/content/monitor/AsyncFileEntry.java

```diff
@@ -0,0 +1,241 @@
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.ConcurrentSkipListSet;
+import javax.annotation.Nullable;
+import javax.validation.constraints.NotNull;
+
+/**
+ * The core of this file is based upon {@link org.apache.commons.io.monitor.FileEntry}, modified to
+ * update only on successful async operations.
+ *
+ * <p>A wrapper class for a Java {@link File} object. This essentially keeps meta-snapshots taken
+ * when commit() is called. hasChanged will compare the current value of the file to the last time
+ * it's been committed, or the snapshot.
+ *
+ * @apiNote once hasChanged returns true, the user must commit the file once it's finished
+ *     processing to create a new meta-snapshot.
+ */
+public class AsyncFileEntry implements Comparable<AsyncFileEntry> {
+
+  //  Only Nullable for GSON serialization
+  @Nullable private final File contentFile;
+  private boolean exists;
+  private long lastModified;
+  private boolean directory;
+  private long length;
+
+  private final Set<AsyncFileEntry> children = new ConcurrentSkipListSet<>();
+
+  //  Leaving transient to avoid loops
+  @Nullable private transient AsyncFileEntry parent;
+
+  private String name;
+
+  public AsyncFileEntry(File file) {
+    this(null, file);
+  }
+
+  public AsyncFileEntry(@Nullable AsyncFileEntry parent, File file) {
+    this.parent = parent;
+    contentFile = file;
+    refresh();
+  }
+
+  //  For GSON serialization
+  private AsyncFileEntry() {
+    contentFile = null;
+  }
+
+  private void refresh() {
+    name = snapName();
+    exists = snapExist();
+    lastModified = snapLastModified();
+    directory = snapDirectory();
+    length = snapLength();
+  }
+
+  public synchronized boolean hasChanged() {
+
+    //  Checks to see if the file has been changed
+    return exists != snapExist()
+        || lastModified != snapLastModified()
+        || directory != snapDirectory()
+        || length != snapLength();
+  }
+
+  public String getName() {
+    return name;
+  }
+
+  public File getFile() {
+    return contentFile;
+  }
+
+  public boolean isDirectory() {
+    return directory;
+  }
+
+  public Optional<AsyncFileEntry> getParent() {
+    return Optional.ofNullable(parent);
+  }
+
+  /**
+   * Checking the network by checking the directory under the file. This works under two
+   * assumptions:
+   *
+   * <ol>
+   *   <li>The root directory will NOT be deleted
+   *   <li>There will NOT be a semlink to another NFS
+   * </ol>
+   */
+  public boolean checkNetwork() {
+    AsyncFileEntry rootParent = parent;
+    while (rootParent != null && rootParent.getParent().isPresent()) {
+      rootParent = rootParent.getParent().get();
+    }
+    if (rootParent == null) {
+      return snapExist();
+    }
+
+    return rootParent.getFile().exists();
+  }
+
+  /**
+   * Tells the AsyncFileEntry that it's successfully been processed and to take a new meta-snapshot
+   */
+  public void commit() {
```

Comment:
✏️ Do you think calling this method `processed` would be clearer?

Reply:
I'm not too sure, I think that `commit()` is fairly clear.

Reply:
`commit()` was pretty clear to me and, I think, even helped my understanding